codebase/patches/mediasourceservice.patch

```diff
@@ -0,0 +1,54 @@
+diff --git a/web/modules/contrib/islandora/src/MediaSource/MediaSourceService.php b/web/modules/contrib/islandora/src/MediaSource/MediaSourceService.php
+index 9399e334..9c160327 100644
+--- a/src/MediaSource/MediaSourceService.php
++++ b/src/MediaSource/MediaSourceService.php
+@@ -281,13 +281,32 @@ class MediaSourceService {
+       }
+ 
+       $directory = $this->fileSystem->dirname($content_location);
+-      if (!$this->fileSystem->prepareDirectory($directory, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS)) {
+-        throw new HttpException(500, "The destination directory does not exist, could not be created, or is not writable");
++      try {
++        if (!$this->fileSystem->prepareDirectory($directory, FileSystemInterface::CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS)) {
++          throw new HttpException(500, "The destination directory does not exist, could not be created, or is not writable");
++        }
++      } catch (\Drupal\Core\Database\IntegrityConstraintViolationException $constraintEx) {
++        // ignore
```

Comment:
What type of error does this suppress?

Reply:
It suppresses `IntegrityConstraintViolationException` that arise when interleaved requests attempt to create a directory with the same name; those errors are benign and shouldn't prevent the creation and attachment of a Media to the Node.